docs/source/examples/pt_zne.md

```diff
@@ -0,0 +1,386 @@
+        )
+
+        for j in range(len(n_qubit_paulis1)):
+            pauli_superop_pauli = np.matmul(
+                n_qubit_paulis1[j], superop_on_pauli_matrix
+            )
+            ptm_matrix[j, i] = (0.5**num_qubits) * np.trace(
+                pauli_superop_pauli
+            )
+
+    return ptm_matrix
+```
+
+Let us consider a simple circuit of a CNOT gate. We are going to subject this circuit to two types of noise and compare
+their respective PTMs.
+
+```{code-cell} ipython3
+import numpy as np
+import seaborn as sns
+import matplotlib.pylab as plt
+
+from cirq import LineQubit, Circuit, CNOT, Ry, depolarize, X, Y, Z
+
+q0 = LineQubit(0)
+q1 = LineQubit(1)
+
+circuit = Circuit(CNOT(q0, q1))
+print(circuit)
+```
+```{code-cell} ipython3
+ptmcnot = ptm_matrix(circuit, 2)
+ax = sns.heatmap(ptmcnot.real, linewidth=0.5, vmin=-1, vmax=1)
+print("Ideal CNOT PTM")
+plt.show()
+```
+```{code-cell} ipython3
+# PTM of a noisy CNOT gate: depolarizing noise
+noisy_circuit_incoherent = circuit.with_noise(depolarize(p=0.3))
+print(noisy_circuit_incoherent)
+
+ptmcnot = ptm_matrix(noisy_circuit_incoherent, 2)
+ax = sns.heatmap(ptmcnot.real, linewidth=0.5, vmin=-1, vmax=1)
+print("\n Pauli Transfer Matrix of noisy CNOT (incoherent)")
+plt.show()
+```
+```{code-cell} ipython3
+# PTM of a noisy CNOT gate: Rz
+noisy_circuit_coherent = circuit.with_noise(Ry(rads=np.pi/12))
+print(noisy_circuit_coherent)
+
+ptmcnot = ptm_matrix(noisy_circuit_coherent, 2)
+ax = sns.heatmap(ptmcnot.real, linewidth=0.5, vmin=-1, vmax=1)
+print("Pauli Transfer Matrix of noisy CNOT (coherent)")
+plt.show()
+```
+If we compare the PTM of the ideal CNOT gate to those when the gate was subjected to incoherent noise and coherent noise, 
+there are additional sources of errors to deal with when coherent noise is acting on the CNOT gate. These can be reduced or tailored to be close to how the incoherent noise PTM appears through Pauli Twirling.
+
+```{code-cell} ipython3
+from mitiq.pt import generate_pauli_twirl_variants
+
+# Generate twirled circuits
+NUM_TWIRLED_VARIANTS = 3
+twirled_circuits = generate_pauli_twirl_variants(
+    circuit, num_circuits=NUM_TWIRLED_VARIANTS)
+print("Example ideal twirled circuit", twirled_circuits[-1], sep="\n")
+```
+Now, lets add coherent noise to the CNOT gate in each twirled circuit.
+```{code-cell} ipython3
+
+noisy_twirled_circuits = []
+
+for circ in twirled_circuits:
+    split_circuit = Circuit(circ[0], circ[1], Ry(rads=np.pi/12)(q0), Ry(rads=np.pi/12)(q1), circ[-1])
+    noisy_twirled_circuits.append(split_circuit)
+
+print("Example noisy twirled circuit", noisy_twirled_circuits[-1], sep="\n")
+```
+
+The twirled PTM is averaged over each noisy twirled circuit such that the new PTM is close to that of the PTM of incoherent noise. We skip the step in this section as we require a very large number of twirled circuits to demonstrate the desired effect of averaging over multiple numpy arrays.
```

Comment:
A bit anti-climatic, I was looking forward to seeing the effect of this on those PTM heatmaps :)

Reply:
Yes, well, this increases the notebook execution time by a lot. The coherent noise that I used is obviously not included in the noisy options allowed by `generate_pauli_twirl_variants`. We should create an issue to clean this portion of pauli twirling. 

So, if I want to generate 100 pauli twirled circuits, I have to splice each of the circuits to make sure the noise is inserted in the correct spot and the pauli twirling gates are also inserted accordingly. Then, the next step is to get their PTM's and sum them up.

Reply:
What if we included an image saying "this is what would happen if you generated the twirled PTM"?

Reply:
I have added 3 examples. 

Ideally, I would prefer this to be a gif to show how the heatmap changes when as the number of pauli twirled circuits is increased. We could keep this as a unitary hackathon issue, if there's not a lot of non-quantum issues.